diff --git a/crypto/init.recovery.qcom_decrypt.rc b/crypto/init.recovery.qcom_decrypt.rc
index 24ee866..7113fd6 100644
--- a/crypto/init.recovery.qcom_decrypt.rc
+++ b/crypto/init.recovery.qcom_decrypt.rc
@@ -88,7 +88,7 @@ service prepdecrypt /system/bin/prepdecrypt.sh
     oneshot
     seclabel u:r:recovery:s0
 
-service qseecomd /system/bin/qseecomd
+service qseecomd /vendor/bin/qseecomd
     user root
     group root
     setenv LD_LIBRARY_PATH /vendor/lib64:/vendor/lib:/system/lib64:/system/lib:/sbin
@@ -104,7 +104,7 @@ service keymaster-3-0 /system/bin/android.hardware.keymaster@3.0-service
     oneshot
     seclabel u:r:recovery:s0
 
-service keymaster-3-0-qti /system/bin/android.hardware.keymaster@3.0-service-qti
+service keymaster-3-0-qti /vendor/bin/hw/android.hardware.keymaster@3.0-service-qti
     user root
     group root
     setenv LD_LIBRARY_PATH /vendor/lib64:/vendor/lib:/system/lib64:/system/lib:/sbin
@@ -136,7 +136,7 @@ service keymaster-4-1 /system/bin/android.hardware.keymaster@4.1-service
     oneshot
     seclabel u:r:recovery:s0
 
-service keymaster-4-1-qti /system/bin/android.hardware.keymaster@4.1-service-qti
+service keymaster-4-1-qti /vendor/bin/hw/android.hardware.keymaster@4.1-service-qti
     user root
     group root
     setenv LD_LIBRARY_PATH /vendor/lib64:/vendor/lib:/system/lib64:/system/lib:/sbin
diff --git a/crypto_fbe/init.recovery.qcom_decrypt.fbe.rc b/crypto_fbe/init.recovery.qcom_decrypt.fbe.rc
index 95f35c8..03bfe00 100644
--- a/crypto_fbe/init.recovery.qcom_decrypt.fbe.rc
+++ b/crypto_fbe/init.recovery.qcom_decrypt.fbe.rc
@@ -37,7 +37,7 @@ service gatekeeper-1-0 /system/bin/android.hardware.gatekeeper@1.0-service
     disabled
     seclabel u:r:recovery:s0
 
-service gatekeeper-1-0-qti /system/bin/android.hardware.gatekeeper@1.0-service-qti
+service gatekeeper-1-0-qti /vendor/bin/hw/android.hardware.gatekeeper@1.0-service-qti
     user root
     group root
     setenv LD_LIBRARY_PATH /vendor/lib64:/vendor/lib:/system/lib64:/system/lib:/sbin
